src/index.js

```diff
@@ -326,6 +308,8 @@ class CalendarHeatmap extends React.Component {
   }
 
   render() {
+    this.valueCache = this.getValueCache(this.props.values);
```

Comment:
@anajavi I don't love tying the cache update to `render` -- is there a reason you implemented it here instead of in `componentDidUpdate` like you brought up in #96 ?

Reply:
I agree that creating cache in render() is not the most elegant thing to do. However, doing it in componentDidUpdate makes it quite complex and causes unnecessary re-render.

Creating the cache in componentDidUpdate requires checking has anything used for cache creation changed in the props, and then calling setState invalidates the render that was just done:
```javascript
  componentDidUpdate(prevProps) {
    if (this.props.classForValue !== prevProps.classForValue ||
      this.props.titleForValue !== prevProps.titleForValue ||
      this.props.endDate !== prevProps.endDate ||
      this.props.startDate !== prevProps.startDate) {
        
      this.setState({
        valueCache: this.getValueCache(this.props.values)
      })
    }
  }
```
And because componentDidUpdate runs after the render, we need to set this.latestProps in there too, which requires checking has anything in the props changed.

Also adding shallow compare update checks changes the way it now works. Currently the cache is created for every render like in this PR.

So doing the cache creation in render() is the simplest way to do it without using getDerivedStateFromProps and it prevents one re-render.

I'm happy to change this if I missed something?

Reply:
React blog seems to suggest memoization with something like [memoize-one](https://github.com/alexreardon/memoize-one):

https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

Reply:
Great points! We discussed this further and given how lightweight memoize-one is, we're okay with adding it- can you update your PR to use it?

Reply:
I've updated the PR to use memoize-one.

I had to pass whole props to it as a parameter, as getValueCache calls other functions which use props. Otherwise the tests would not pass.

Also I had to add @babel/plugin-proposal-class-properties to be able to use the same syntax as in the react-blog.

Do you think the memoize-one should be declared as external in rollup.config.js?

Reply:
Is there something more to be done?